--- a/scons/simula-scons/simula_scons/pkgconfiggenerators/armadillo.py
+++ b/scons/simula-scons/simula_scons/pkgconfiggenerators/armadillo.py
@@ -101,6 +101,13 @@
             cmdstr = "%s %s -o a.out %s" % \
                      (linker, cpp_file.replace('.cpp', '.o'), libs)
             linkFailed, cmdoutput = getstatusoutput(cmdstr)
+            if linkFailed:
+                # try adding -llapack to get around Arch Linux "undefined
+                # reference to dgels_" problem
+                libs += " -llapack"
+                cmdstr = "%s %s -o a.out %s" % \
+                         (linker, cpp_file.replace('.cpp', '.o'), libs)
+                linkFailed, cmdoutput = getstatusoutput(cmdstr)
             if not linkFailed:
                 break
         else:
