*** a/scons/simula-scons/simula_scons/pkgconfiggenerators/boost_program_options.py	2010-02-17 17:50:50.000000000 +0000
--- b/scons/simula-scons/simula_scons/pkgconfiggenerators/boost_program_options.py	2010-05-18 01:02:56.372878104 +0000
***************
*** 60,73 ****
    lib_dir = os.path.join(boost.getBoostDir(sconsEnv=sconsEnv), 'lib')
    po_lib = "boost_program_options"
    app = os.path.join(os.getcwd(), "a.out")
!   cmdstr = "%s -o %s -L%s -l%s %s" % \
!            (linker, app, lib_dir, po_lib, cpp_file.replace('.cpp', '.o'))
    linkFailed, cmdoutput = getstatusoutput(cmdstr)
    if linkFailed:
      # try to append -mt to lib
      po_lib += "-mt"
!     cmdstr = "%s -o %s -L%s -l%s %s" % \
!              (linker, app, lib_dir, po_lib, cpp_file.replace('.cpp', '.o'))
      linkFailed, cmdoutput = getstatusoutput(cmdstr)
      if linkFailed:
        remove_cppfile(cpp_file, ofile=True)
--- 60,73 ----
    lib_dir = os.path.join(boost.getBoostDir(sconsEnv=sconsEnv), 'lib')
    po_lib = "boost_program_options"
    app = os.path.join(os.getcwd(), "a.out")
!   cmdstr = "%s  %s -o %s -L%s -l%s" % \
!            (linker, cpp_file.replace('.cpp', '.o'), app, lib_dir, po_lib)
    linkFailed, cmdoutput = getstatusoutput(cmdstr)
    if linkFailed:
      # try to append -mt to lib
      po_lib += "-mt"
!     cmdstr = "%s %s -o %s -L%s -l%s" % \
!              (linker, cpp_file.replace('.cpp', '.o'), app, lib_dir, po_lib)
      linkFailed, cmdoutput = getstatusoutput(cmdstr)
      if linkFailed:
        remove_cppfile(cpp_file, ofile=True)
diff -c a/scons/simula-scons/simula_scons/pkgconfiggenerators/libxml.py b/scons/simula-scons/simula_scons/pkgconfiggenerators/libxml.py
*** a/scons/simula-scons/simula_scons/pkgconfiggenerators/libxml.py	2010-02-17 17:50:50.000000000 +0000
--- b/scons/simula-scons/simula_scons/pkgconfiggenerators/libxml.py	2010-05-18 01:05:36.562861966 +0000
***************
*** 46,52 ****
                                         program=cpp_test_version_str,
                                         errormsg=cmdoutput)
  
!     cmdstr = "%s -o a.out %s %s" % (linker, libs, cppfile.replace('.cpp', '.o'))
      linkFailed, cmdoutput = getstatusoutput(cmdstr)
      if linkFailed:
          remove_cppfile(cppfile, ofile=True)
--- 46,52 ----
                                         program=cpp_test_version_str,
                                         errormsg=cmdoutput)
  
!     cmdstr = "%s %s -o a.out %s" % (linker, cppfile.replace('.cpp', '.o'), libs)
      linkFailed, cmdoutput = getstatusoutput(cmdstr)
      if linkFailed:
          remove_cppfile(cppfile, ofile=True)
***************
*** 230,237 ****
                                         program=cpp_test_lib_str,
                                         errormsg=cmdoutput)
  
!     cmdstr = "%s -o a.out %s %s" % \
!              (linker, libs, cpp_file.replace('.cpp', '.o'))
      linkFailed, cmdoutput = getstatusoutput(cmdstr)
      if linkFailed:
          remove_cppfile(cpp_file, ofile=True)
--- 230,237 ----
                                         program=cpp_test_lib_str,
                                         errormsg=cmdoutput)
  
!     cmdstr = "%s %s -o a.out %s" % \
!              (linker, cpp_file.replace('.cpp', '.o'), libs)
      linkFailed, cmdoutput = getstatusoutput(cmdstr)
      if linkFailed:
          remove_cppfile(cpp_file, ofile=True)
